src/v2/System/useSystemContext.ts

```diff
@@ -6,5 +6,5 @@ import { SystemContext } from "./SystemContext"
  */
 export const useSystemContext = () => {
   const systemContext = useContext(SystemContext)
-  return systemContext
+  return { ...systemContext, isLoggedIn: !!systemContext.user }
```

Comment:
We move this to the `SystemContextProvider` component where the rest of these initializers live

Reply:
This creates a bunch of annoying type errors. Doing a computed prop like this is fine IMO.

Reply:
I can weaken the type of course; which is I guess why all of the other fields are optional.

Reply:
What I mean is:

![](https://static.damonzucconi.com/_capture/nN2Thxwi.png)

This kind of thing bothers me since there's no way this is actually undefined — in practice... I guess this is fine since you're typically only using it for a switch and the undefined would be falsey anway. Three-valued booleans just weird me out a bit.

Reply:
If you feel the call to fix those types, they need it, but we should def be putting this stuff with the rest of the bunch rather than introducing drift